src/Backend/network.py

```diff
@@ -32,11 +32,15 @@ def __init__(self, *args, **kwargs):
 
     def deepcopy(self):
         network_copy = copy.deepcopy(self)
-        network_copy._loads_index = self._loads_index
+        if hasattr(self, '_loads_index'):
+            setattr(network_copy, '_loads_index', self._loads_index)
```

Comment:
this does not appear to be a "deep" copy (not even a copy). It's the copy of the reference to the _loads_index I suppose (though i'm not sure and assume here _loads_index is a complex type, eg list, dictionary etc.)
To be sure to make a copy, one could do:
> setattr(network_copy, '_loads_index', copy.deepcopy(self._loads_index))